syntax = "proto3";

service Booking {
    rpc GetBookings(UserID) returns (BookingResponse) {}
    rpc GetListBookings(Empty1) returns (stream BookingResponse) {}
    rpc AddBooking(AddBookingRequest) returns (BookingResponse) {}
}

message UserID {
    string userid = 1;
}

message AddBookingRequest {
    string userid = 1;
    string date = 2;
    string movieid = 3;
}

message Dates {
    string date = 1;
    repeated string movie = 2;
}

message Booking_info {
    string userid = 1;
    repeated Dates dates = 2;
}

message BookingResponse {
    Booking_info booking = 1;
}

message Empty1 {
}


